BlackMarket/app/Extensions/View+extension.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  View+extension.swift
+//  BlackMarket
+//
+//  Created by Tarek Radovan on 15/12/2022.
+//
+
+import Foundation
+import SwiftUI
+
+//To manage @FocusedState in TCA 
+extension View {
+  func synchronize<Value>(
+    _ first: Binding<Value>,
+    _ second: FocusState<Value>.Binding
+  ) -> some View {
+    self
+      .onChange(of: first.wrappedValue) { second.wrappedValue = $0 }
+      .onChange(of: second.wrappedValue) { first.wrappedValue = $0 }
```

Comment:
do you know if this causes a reentrancy anomaly in the signal? like one change causes the other to change and so on, infinitely.

Reply:
It doesn't ... I tested with debugging and The  focusState didn't change when the value was reintroduced because it was the same... I think that is because the hashable comparison